x/btcstaking/types/create_delegation_parser.go

```diff
@@ -104,20 +104,20 @@ type ParsedProofOfInclusion struct {
 }
 
 func NewParsedProofOfInclusion(
-	info *btcckpttypes.TransactionInfo,
+	ip *InclusionProof,
 ) (*ParsedProofOfInclusion, error) {
-	if info == nil {
-		return nil, fmt.Errorf("cannot parse nil *btcckpttypes.TransactionInfo")
+	if ip == nil {
```

Comment:
I think we need to allow `nil` inclusion proof as  now this proof is optional in `MsgCreateDelegation`

Also in the struct `ParsedCreateDelegationMessage` I would leave doc comment for `StakingTxProofOfInclusion` field that, so something along the lines:
```
// StakingTxInclusionProof is optional is and it is up to the caller to verify
// whether it is present or not
StakingTxProofOfInclusion  *ParsedProofOfInclusion
```